geoips/commandline/test_interfaces.py

```diff
@@ -53,21 +51,107 @@ def main():
         )
         try:
             out_dict = test_curr_interface()
+            out_dicts[curr_interface] = out_dict
         except Exception:
             print(traceback.format_exc())
             raise
 
-        print(f"SUCCESSFUL INTERFACE {curr_interface}")
-
         ppprinter = pprint.PrettyPrinter(indent=2)
         ppprinter.pprint(out_dict)
 
+    for intname in out_dicts:
+        failed_one = False
+        out_dict = out_dicts[intname]
         for modname in out_dict["validity_check"]:
             if not out_dict["validity_check"][modname]:
-                print(f"FAILED INTERFACE {curr_interface} on {modname}")
-                raise TypeError(
-                    f"Failed validity check on {modname} in interface {curr_interface}"
-                )
+                failed_plugins += [f"{intname} on {modname}"]
+                failed_one = True
+            else:
+                successful_plugins += [f"{intname} on {modname}"]
+        if not failed_one:
+            successful_interfaces += [intname]
+
+    return failed_plugins, successful_interfaces, successful_plugins
+
+
+def main():
+    """Script to test all dev and stable interfaces."""
+
+    failed_interfaces = []
+    failed_plugins = []
+    successful_interfaces = []
+    successful_plugins = []
+
+    # Test all the "old" interfaces using the original logic.
+    # (
+    #     failed_plugins,
+    #     successful_interfaces,
+    #     successful_plugins,
+    # ) = test_deprecated_interfaces(
+    #     failed_plugins, successful_interfaces, successful_plugins
+    # )
+
+    interfaces = [
+        "algorithms",
+        "colormaps",
+        "filename_formats",
+        "interpolators",
+        "output_formats",
+        "procflows",
+        "readers",
+        "title_formats",
+    ]
```

Comment:
```suggestion
    interfaces = [
        "algorithms",
        "colormaps",
        "filename_formats",
        "interpolators",
        "output_formats",
        "procflows",
        "readers",
        "title_formats",
    ]
```

Should this just be determined by what exists in `interfaces`? Something like:
```
interfaces = [inter for inter in geoips.interfaces if isinstance(inter, geoips.interfaces.BaseInterface) and type(inter) not geoips.interfaces.BaseInterface]
```

If so, I'll make the change. There are a couple little details to take care of here.

Reply:
Sure, you can make that change

Reply:
I will resolve this in a separate PR. This is more involved than I thought.